alpaca_trade_api/rest.py

```diff
@@ -562,21 +562,31 @@ def _data_get(self,
             data = kwargs
             data['limit'] = actual_limit
             data['page_token'] = page_token
-            resp = self.data_get(
-                '/{}/{}/{}'.format(endpoint_base, symbol, endpoint),
-                data=data, api_version=api_version)
-            items = resp.get(endpoint, []) or []
-            for item in items:
-                yield item
-            total_items += len(items)
+            if isinstance(symbol, str):
+                path = f'/{endpoint_base}/{symbol}/{endpoint}'
+            else:
+                path = f'/{endpoint_base}/{endpoint}'
+                data['symbols'] = ','.join(symbol)
+            resp = self.data_get(path, data=data, api_version=api_version)
+            if isinstance(symbol, str):
+                for item in resp.get(endpoint, []) or []:
+                    yield item
+                    total_items += 1
+            else:
+                by_symbol = resp.get(endpoint, {}) or {}
+                for sym, items in sorted(by_symbol.items()):
+                    for item in items or []:
+                        item['S'] = sym
+                        yield item
+                        total_items += 1
```

Comment:
so now if we limit to 1000 and we ask for 1000 symbols we get 1 item for each? shouldn't it be 1000 for each symbol?

Reply:
No. The server-side multi symbol endpoints return the requested data in symbol-order first, then by timestamp. So for if we ask for 1000 symbols and set the limit to 1000, we may only get from the first symbol (if it has 1000 items in the given interval). Btw unfortunately there is a dissonance between the REST API's `limit` parameter and the SDK's `limit` parameter: the first one is actually a page size, while the second one is the actual total limit of the (automatically paginated) response size.